### PR TITLE
Document `custom_modules` build option

### PR DESCRIPTION
**Pull request by Xrayez**
Describes the feature introduced in godotengine/godot#36922.

CC @RaTcHeT302 https://github.com/godotengine/godot-proposals/issues/565#issuecomment-633647029. 🙂

### CODE REVIEW

File: development/cpp/custom_modules_in_cpp.rst [hunk 1]
```diff
@@ -257,6 +257,38 @@ The output will be ``60``.
              template. See the :ref:`Compiling <toc-devel-compiling>` pages
              for more information.
 
+Compiling a module externally
+-----------------------------
+
+Lets take our "summator" module and move it to the engine's parent directory:
```
Comment (akien-mga) [inline development/cpp/custom_modules_in_cpp.rst:263]: I'd add a sentence to explain the rationale. As is as a reader, I'm not sure why I would be asked to do this after building my module and learning how to use it.

Comment (Xrayez) [reply development/cpp/custom_modules_in_cpp.rst]: Added not a sentence, but a paragraph! The rationale might be a bit opinionated but I take this from my own experience.

File: development/compiling/introduction_to_the_buildsystem.rst [hunk 2]
```diff
@@ -182,6 +182,36 @@ This flag appends ``.32`` or ``.64`` suffixes to resulting binaries when
 relevant. If ``bits=default`` is used, the suffix will match the detected
 architecture.
 
+.. _doc_buildsystem_custom_modules:
+
+Custom modules
+--------------
+
+It's possible to compile modules residing outside of the Godot's directory
```
Comment (mhilbrunner) [inline development/compiling/introduction_to_the_buildsystem.rst:190]:
```suggestion
It's possible to compile modules residing outside of Godot's directory
```

File: development/compiling/introduction_to_the_buildsystem.rst [hunk 3]
```diff
@@ -182,6 +182,36 @@ This flag appends ``.32`` or ``.64`` suffixes to resulting binaries when
...
+
+::
+
+    scons custom_modules="../modules,/abs/path/to/modules,~/src/godot_modules"
+
+.. note::
+
+    If there's any custom module with the exact directory name as a built-in
+    module, the engine will only compile the custom one. This logic can be used
+    to override the built-in module's implementation altogether.
```
Comment (mhilbrunner) [inline development/compiling/introduction_to_the_buildsystem.rst:209]:
```suggestion
    module, the engine will only compile the custom one. This logic can be used
    to override built-in module implementations.
```

File: development/cpp/custom_modules_in_cpp.rst [hunk 4]
```diff
@@ -257,6 +257,57 @@ The output will be ``60``.
...
+
+Compiling a module involves moving the module's sources directly under the
+engine's ``modules/`` directory. While this is the most straightforward way to
+compile a module, there are a couple of reasons as to why this might not be a
+practical thing to do:
+
+1. Having to manually copy modules sources every time you want to compile the
+   engine with or without the module, or taking additional steps needed to
+   manually disable a module during compilation with a build option similar to
+   ``module_summator_enabled=no``. Making symbolic links may also be a solution,
```
Comment (mhilbrunner) [inline development/cpp/custom_modules_in_cpp.rst:271]:
```suggestion
   ``module_summator_enabled=no``. Creating symbolic links may also be a solution,
```

**Comment by mhilbrunner** [comment]
Thank you!

File: development/compiling/introduction_to_the_buildsystem.rst (hunk 3 above)
Comment (Xrayez) [reply development/compiling/introduction_to_the_buildsystem.rst]: I have a real hard time understanding `the`s, that's certainly not a built-in feature of my language, thanks. 😛

**Comment by aaronfranke** [comment]
@Xrayez Can you add documentation for how to include files from Godot in a custom module not in Godot's modules folder? `#include "core/reference.h"` etc doesn't seem to work as it can't find the file.

**Pull request merged**